mods/gen3/random-teams.js

```diff
@@ -98,46 +98,56 @@ class RandomGen3Teams extends RandomGen4Teams {
 				case 'focuspunch':
 					if (!hasMove['substitute'] || counter.damagingMoves.length < 2) rejected = true;
 					break;
+				case 'meanlook': case 'spiderweb':
+					if (!hasMove['batonpass'] && !hasMove['perishsong']) rejected = true;
+					break;
 				case 'perishsong':
 					if (!hasMove['meanlook']) rejected = true;
 					break;
-				case 'rest': {
+				case 'rest':
 					if (movePool.includes('sleeptalk')) rejected = true;
 					break;
-				}
 				case 'sleeptalk':
 					if (!hasMove['rest']) rejected = true;
 					if (movePool.length > 1) {
 						let rest = movePool.indexOf('rest');
 						if (rest >= 0) this.fastPop(movePool, rest);
 					}
 					break;
+				case 'sunnyday':
+					if (!hasMove['solarbeam']) rejected = true;
+					break;
 
 				// Set up once and only if we have the moves for it
 				case 'bellydrum': case 'bulkup': case 'curse': case 'dragondance': case 'swordsdance':
 					if (counter.setupType !== 'Physical' || counter['physicalsetup'] > 1) rejected = true;
 					if (counter.Physical + counter['physicalpool'] < 2 && !hasMove['batonpass'] && (!hasMove['rest'] || !hasMove['sleeptalk'])) rejected = true;
 					isSetup = true;
 					break;
-				case 'calmmind': case 'growth': case 'nastyplot': case 'tailglow':
+				case 'calmmind': case 'growth': case 'tailglow':
 					if (counter.setupType !== 'Special' || counter['specialsetup'] > 1) rejected = true;
 					if (counter.Special + counter['specialpool'] < 2 && !hasMove['batonpass'] && (!hasMove['rest'] || !hasMove['sleeptalk'])) rejected = true;
 					isSetup = true;
 					break;
-				case 'agility': case 'rockpolish':
+				case 'agility':
 					if (counter.damagingMoves.length < 2 && !hasMove['batonpass']) rejected = true;
 					if (hasMove['rest'] && hasMove['sleeptalk']) rejected = true;
 					if (!counter.setupType) isSetup = true;
 					break;
-				case 'flail':
+				case 'endeavor': case 'flail': case 'reversal':
 					if (!hasMove['substitute'] && !hasMove['endure']) rejected = true;
 					break;
 
 				// Bad after setup
-				case 'explosion':
+				case 'destinybond': case 'explosion': case 'memento': case 'selfdestruct':
 					if (counter.setupType || !!counter['recovery'] || hasMove['rest']) rejected = true;
+					if (moveid === 'destinybond' && (hasMove['explosion'] || hasMove['selfdestruct'])) rejected = true;
+					if (moveid === 'memento' && hasMove['destinybond']) rejected = true;
 					break;
-				case 'foresight': case 'protect': case 'roar':
+				case 'protect':
+					if (!hasAbility['Speed Boost'] && !hasMove['perishsong'] && !hasMove['toxic'] && !hasMove['wish']) rejected = true;
+					break;
+				case 'roar': case 'whirlwind':
 					if (counter.setupType && !hasAbility['Speed Boost']) rejected = true;
```

Comment:
Not sure why having Speed boost would help roar/whirlwind as they are both -6 priority.

Reply:
Good catch, I missed taking that out when I split Protect into its own check.